### ISSUE TITLE
`AssignmentInCondition` didn't find the style problem

### ISSUE THREAD

**Issue by itsalongstory** [rubocop/rubocop#13446]
## Expected behavior

RuboCop should find the style problem in the `if` statement

## Actual behavior

RuboCop didn't find the style problem

## Steps to reproduce the problem

test.rb

```rb
var = 1

if (
  (var == 1) ||
  (var = '1')
)
  puts "it's 1"
end
```

Run:

```bash
ruby test.rb
```

Result:

```
test.rb:5: warning: found `= literal' in conditional, should be ==
it's 1
```

Run:

```bash
rubocop --only Lint/AssignmentInCondition -a test.rb
```

Result:

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```bash
root@localhost:~$ rubocop -v
1.68.0
```
## Real Business Code

![image](https://github.com/user-attachments/assets/d84ea312-4321-4e88-a5be-9045bc4f2e42)

**Comment by dvandersluis** [comment]
There's no offense expected from `Lint/AssignmentInCondition` here, when `AllowSafeAssignment` is `true`.

`order.transport_type_value = '1' && order.ie_type_value == '0'` is a single assignment, equivalent to `order.transport_type_value = ('1' && order.ie_type_value == '0')`, and therefore the entire clause is parenthesized, and thus acceptable.

**Comment by itsalongstory** [comment]
> There's no offense expected from `Lint/AssignmentInCondition` here, when `AllowSafeAssignment` is `true`.

Thanks for explain.
